python/grass/__init__.py

```diff
@@ -75,16 +93,48 @@ def _translate(text):
             # does not raise an exception even if the locale settings is broken
             # or the translation files were not installed.
             fallback = True
-            translation = gettext.translation(
-                "grasslibs", locale_dir, fallback=fallback
-            )
-            # Add other domains as fallback.
-            translation.add_fallback(
-                gettext.translation("grassmods", locale_dir, fallback=fallback)
-            )
-            translation.add_fallback(
-                gettext.translation("grasswxpy", locale_dir, fallback=fallback)
-            )
+            translation = None
+            domains = ("grasslibs", "grassmods", "grasswxpy")
+
+            def apply_trans(kwargs):
+                nonlocal translation
+                if kwargs["domain"] == "grasslibs":
+                    translation = gettext.translation(**kwargs)
+                else:
+                    translation.add_fallback(gettext.translation(**kwargs))
+
+            try:
+                # Get user defined lang
+                lang = _UserSettings.Get(
+                    group="language",
+                    key="locale",
+                    subkey="lc_all",
+                )
+            except NameError:
+                # If _UserSettings isn't imported during tests (without wxGUI)
+                lang = None
+
+            if lang and lang not in ("en", "system"):
```

Comment:
Please add a comment on why these two need a special treatment.

Reply:
+1 for a comment. The rationale – we do not ship "en" or "system" translations and thus they need a different logic in setting up.

Reply:
Yes you're right in context of existed translation PO/MO files. But user has option via GUI settings change language to "en" (which is the language if no translation is used) or "system" (OS locale language) language and is justified.

A. Use case:

1. User has system with "cz_CZ.utf8" locale settings (wxGUI start with Czech language)
2. User want wxGUI with English language (choose "en" option from wxGUI Settings dialog Choose language combobox  widget)

B. Use case

1. User has system with "cz_CZ.utf8" locale settings (wxGUI start with Czech language)
2. User want wxGUI with English language (choose "en" option from wxGUI Settings dialog Choose language combobox  widget)
3. Later user want use wxGUI with default system language (default system locale which is "cz_CZ.utf8", choose "system" option from wxGUI Settings dialog Choose language combobox widget)